py3status/modules/xrandr_rotate.py

```diff
@@ -62,17 +62,17 @@ def post_config_hook(self):
         self.displayed = ''
 
     def _get_all_outputs(self):
-        cmd = 'xrandr -q | grep " connected [^(]" | cut -d " " -f1'
-        return self.py3.command_output(cmd, shell=True).splitlines()
+        outputs = self.py3.command_output(['xrandr']).splitlines()
+        return [x.split()[0] for x in outputs if ' connected' in x]
 
     def _get_current_rotation_icon(self, all_outputs):
         output = self.screen or all_outputs[0]
-        cmd = 'xrandr -q | grep "^' + output + '" | cut -d " " -f4'
-        output = self.py3.command_output(cmd, shell=True).strip()
-        # xrandr may skip printing the 'normal', in which case the output would
-        # start from '('
-        is_horizontal = (output.startswith('(') or
-                         output in ['normal', 'inverted'])
+        data = self.py3.command_output(['xrandr']).splitlines()
+        orientation = ''.join(
+            [x.split()[3] for x in data if x.startswith(output)])
+        # xrandr may skip 'normal' so we check if it starts with '(' too
```

Comment:
this comment removes information that was in the previous one.

The logic is also changed.  I think using command_output is good but we should 100% do the same as we were before to prevent bugs.  If there are additional bugs they should be fixed separatly so they can be understood fully

Reply:
I think this comment is incorrect based on me playing around with `xrandr`. Anyhow, if issues appears for some users, I am happy to address them.

>The logic is also changed. I think using command_output is good but we should 100% do the same as we were before to prevent bugs. If there are additional bugs they should be fixed separatly so they can be understood fully

This PR. It does 100% same thing, minus the comment. The logic is same. The `fix orientation` changes how we parse using loop instead of one-time thing... so I keep/fix them separated.

Reply:
Je ne comprend pas. C'est important vous sont le communiqué simplement. Vous dit le `fix orientation` mais pour moi, je ne sais pas le explination.

Je m'excuse mon explanation mais je pense c'est similar à vôtre mots. 

C'est possible je suis, comme vous dit  'a no f'ing conception de la issue'

Reply:
Plus tôt, il y avait des différences entre toutes les sorties et les sorties actives. Tous mes moniteurs sont actifs, donc il est apparu parfait ici. La différence se produit lorsque vous désactivez / activez un affichage, ce qui est ce qu'il faisait à chaque fois qu'il docks son ordinateur portable. Je renomme la méthode de tous en active pour explicitement rendre plus clair que ce n'est pas une f'ing toutes les sorties.

Le fix oritentation corrige l'analyse incorrecte qui se produit (icône verticale) lorsque nous obtenons une mauvaise sortie de xrandr telle que 'primary' (il compense les champs) ou des lignes vides. -- Google Translate